GoogleForCreators/web-stories-wp
# Dashboard: Incorporate the rest of the filters into FiltersProvider

## Context


Move all filters/sorting logic into one place.

## Summary


Incorporates the 'status', 'keyword search', and 'sort' in the FiltersProvider.

## Relevant Technical Choices


Moving all the filters/sort into one place and using a hook to tap into them is more maintainable than prop drilling.

## To-do


The way we use 'tracking' when 'searching' may need to be updated now that the logic has been moved.  Possibly more refactoring to remove old/unnecessary logic.

## User-facing changes



## Testing Instructions






This PR can be tested by following these steps:

1. Go to the Dashboard
2. Test all filters and sort on the Grid View and the List View
- Notice these should be working: Taxonomy filters, author filter, status filter, keyword search, and sort
3. Also test #11668 - this ticket deals with that issue as well


## Reviews

### Does this PR have a security-related impact?
No


### Does this PR change what data or activity we track or use?

Possibly - I do believe we'll need to update the search tracking

### Does this PR have a legal-related impact?
No


## Checklist




---



Fixes #11668
Fixes #11789 
Fixes #11780
